input-output-hk/lace
# feature(staking): add selection management buttons

# Checklist

- [x] JIRA - [\<link>](https://input-output.atlassian.net/browse/LW-7326)

---

## Proposed solution

 - New buttons on the pool details drawer for managing multi-staking selections
 - Basket component appearing when there are selections
 - "Clear" button of the basket is now functional

## Testing

1. Build lace with new multi staking
2. Go to staking/browse pools
3. Click on a pools and use new buttons
4. Check a case when we are already staking on a pool (need to stake in old UI first)

## Screenshots

Basket with selections
<img width="1253" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/43defa61-7e74-432e-a80f-1a1fadc8170d">
Pool which we currently staking on
<img width="1253" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/6d5197f0-f293-4452-a36f-aa6f63965bfc">
Pool we are NOT currently staking on + there is no multi-staking selections yet
<img width="1253" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/cdec5e21-46f1-4d75-aabb-dc4c801a675a">
Pool not selected for multi-staking + there is already one pool selected
<img width="1253" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/d29facea-24df-4c83-abe7-0f11ae80a959">
Pool selected for multi-staking
<img width="1253" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/8c484e11-8079-4c49-8b5b-14a8e0bf0965">
Pool not selected for multi-staking + already selected maximum amount of 5
<img width="1253" alt="image" src="https://github.com/input-output-hk/lace/assets/9270461/3aed3ec3-0071-441e-9cac-76d48062754b">






---
# Allure report
`allure-report-publisher` generated test report!



**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@d1terqjryk7mu9.cloudfront.net/linux/chrome/830/5472729917/index.html) for [3b082401](https://github.com/input-output-hk/lace/pull/221/commits/3b082401da568d4d0c10ba55b9130858fcda7016)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 36     | 0      | 0       | 0     | 36    | ✅     |
